browser/loaders/ComponentLoader.js

```diff
@@ -33,11 +34,16 @@ class ComponentLoader extends LoaderBase {
 		this._eventBus = locator.resolve('eventBus');
 
 		/**
-		 * Current template provider.
-		 * @type {TemplateProvider}
+		 * Current template provider map.
+		 * @type {Object}
 		 * @private
 		 */
-		this._templateProvider = locator.resolve('templateProvider');
+		this._templateProvidersByNames = templateHelper
+			.resolveTemplateProviders(locator)
+			.reduce((map, current) => {
+				map[current.getName()] = current;
+				return map;
+			}, Object.create(null));
```

Comment:
Perhaps we have to realise it in `templateHelper` as `resolveTemplateProvidersByNames()`